src/Calculations.ts

```diff
@@ -4,5 +4,46 @@ import { Point3D } from './DataTypes';
 /* Compute the angle between the first three Point3Ds in the provided Array
  * in degrees, with index 0 as theta. */
 export function computeAngle(points: Array<Point3D|null>): number {
-    return 178.600363111;
+
+    if (points.length < 3)
+        return NaN;
+
+    const first3HasNullOrNaN = points.slice(0,3)
+        //map each element to a boolean (true indicates bad data)
+        .map(point => point===null||isNaN(point.x)||isNaN(point.y)||isNaN(point.z))
```

Comment:
these two calls can be replaced with just `some`

Reply:
Ah, I forgot about `some`. Fixed